doc/source/blueprint_cce_vault.rst

```diff
@@ -0,0 +1,592 @@
+         echo "ssl.quorum.trustStore.location=/tls/ca.pem" >> $ZK_CONFIG_FILE
+       fi
+       for (( i=1; i<=$ZK_REPLICAS; i++ ))
+       do
+           echo "server.$i=$NAME-$((i-1)).$DOMAIN:$ZK_SERVER_PORT:$ZK_ELECTION_PORT" >> $ZK_CONFIG_FILE
+       done
+       rm -f $LOG4J_PROPERTIES
+       echo "zookeeper.root.logger=$ZK_LOG_LEVEL, CONSOLE" >> $LOG4J_PROPERTIES
+       echo "zookeeper.console.threshold=$ZK_LOG_LEVEL" >> $LOG4J_PROPERTIES
+       echo "zookeeper.log.threshold=$ZK_LOG_LEVEL" >> $LOG4J_PROPERTIES
+       echo "zookeeper.log.dir=$ZK_DATA_LOG_DIR" >> $LOG4J_PROPERTIES
+       echo "zookeeper.log.file=zookeeper.log" >> $LOG4J_PROPERTIES
+       echo "zookeeper.log.maxfilesize=256MB" >> $LOG4J_PROPERTIES
+       echo "zookeeper.log.maxbackupindex=10" >> $LOG4J_PROPERTIES
+       echo "zookeeper.tracelog.dir=$ZK_DATA_LOG_DIR" >> $LOG4J_PROPERTIES
+       echo "zookeeper.tracelog.file=zookeeper_trace.log" >> $LOG4J_PROPERTIES
+       echo "log4j.rootLogger=\${zookeeper.root.logger}" >> $LOG4J_PROPERTIES
+       echo "log4j.appender.CONSOLE=org.apache.log4j.ConsoleAppender" >> $LOG4J_PROPERTIES
+       echo "log4j.appender.CONSOLE.Threshold=\${zookeeper.console.threshold}" >> $LOG4J_PROPERTIES
+       echo "log4j.appender.CONSOLE.layout=org.apache.log4j.PatternLayout" >> $LOG4J_PROPERTIES
+       echo "log4j.appender.CONSOLE.layout.ConversionPattern=%d{ISO8601} [myid:%X{myid}] - %-5p [%t:%C{1}@%L] - %m%n" >> $LOG4J_PROPERTIES
+       if [ -n "$JMXDISABLE" ]
+       then
+           MAIN=org.apache.zookeeper.server.quorum.QuorumPeerMain
+       else
+           MAIN="-Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=$JMXPORT -Dcom.sun.management.jmxremote.authenticate=$JMXAUTH -Dcom.sun.management.jmxremote.ssl=$JMXSSL -Dzookeeper.jmx.log4j.disable=$JMXLOG4J org.apache.zookeeper.server.quorum.QuorumPeerMain"
+       fi
+       set -x
+       exec java -cp "$CLASSPATH" $JVMFLAGS $MAIN $ZK_CONFIG_FILE
+
+     vault-agent-config.hcl: |
+       exit_after_auth = true
+       pid_file = "/home/vault/pidfile"
+       auto_auth {
+           method "kubernetes" {
+               mount_path = "auth/kubernetes_cce1"
+               config = {
+                   role = "zookeeper"
+                   token_path = "/run/secrets/tokens/vault-token"
+               }
+           }
+           sink "file" {
+               config = {
+                   path = "/home/vault/.vault-token"
+               }
+           }
+       }
+
+       cache {
+           use_auto_auth_token = true
+       }
+
+       # ZK is neat-picky on cert file extensions
+       template {
+         destination = "/tls/ca.pem"
+         contents = <<EOT
+       {{- with secret "/secret/data/tls/ca" }}{{ .Data.data.certificate }}{{ end }}
+       EOT
+       }
+
+       template {
+         destination = "/tls/server/server.pem"
+         contents = <<EOT
+       {{- with secret "{{ zookeeper.vault_server_cert_path }}" }}{{ .Data.data.certificate }}
```

Comment:
you kept here ansible variable references...i think it would be good to rename it to real path instead